controller/deployments.go

```diff
@@ -441,17 +441,74 @@ func (c *DeploymentsController) WatchEnvironmentEventsWSHandler(ctx *app.WatchEn
 				}
 				return
 			}
-			err = websocket.JSON.Send(ws, item)
+
+			eventItem, err := transformItem(item)
 			if err != nil {
-				log.Error(ctx, map[string]interface{}{
-					"err": err,
-				}, "error sending events")
-				return
+				sendWebsocketJSON(ctx, ws, map[string]interface{}{"error": "unable to parse Kubernetes event"})
+			} else {
+				err = websocket.JSON.Send(ws, eventItem)
+				if err != nil {
+					log.Error(ctx, map[string]interface{}{
+						"err": err,
+					}, "error sending events")
+					return
+				}
 			}
 		}
 	}
 }
 
+//DeploymentsEvent is the transformed Kubernetes v1.Event item
+type DeploymentsEvent struct {
+	// The object that this event is about.
+	InvolvedObject v1.ObjectReference `json:"involvedObject" protobuf:"bytes,2,opt,name=involvedObject"`
+
+	// This should be a short, machine understandable string that gives the reason
+	// for the transition into the object's current status.
+	// TODO: provide exact specification for format.
+	// +optional
+	Reason string `json:"reason,omitempty" protobuf:"bytes,3,opt,name=reason"`
+
+	// A human-readable description of the status of this operation.
+	// TODO: decide on maximum length.
+	// +optional
+	Message string `json:"message,omitempty" protobuf:"bytes,4,opt,name=message"`
+
+	// The number of times this event has occurred.
+	// +optional
+	Count int32 `json:"count,omitempty" protobuf:"varint,8,opt,name=count"`
+
+	// Type of this event (Normal, Warning), new types could be added in the future
+	// +optional
+	Type string `json:"type,omitempty" protobuf:"bytes,9,opt,name=type"`
+
+	// CreationTimestamp is a timestamp representing the server time when this object was
+	// created. It is not guaranteed to be set in happens-before order across separate operations.
+	// Clients may not set this value. It is represented in RFC3339 form and is in UTC.
+	//
+	// Populated by the system.
+	// Read-only.
+	// Null for lists.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	CreationTimestamp metaV1.Time `json:"creationTimestamp,omitempty" protobuf:"bytes,8,opt,name=creationTimestamp"`
```

Comment:
Perhaps you can use a Field Selector to limit the included fields, instead of having to transform the data yourself? https://docs.openshift.com/online/rest_api/api/v1.Event.html#Get-api-v1-watch-namespaces-namespace-events

Reply:
My understanding is that FieldSelector limits the set of objects returned based by some conditions on the object fields, e.g. return all events with field `X` with value `Y`. Here we're modifying the returned object to have a subset of fields with the set of objects being unchanged. Does that make sense?

Reply:
You're correct. The documentation for field selectors is lacking...